### ISSUE TITLE
docker run, add restart option "limited"

### ISSUE THREAD

**Issue by pier4r** [moby/moby#47801]
### Description

The documentation as today (2024-05-06) is

```
    --restart=no
    --restart=always
    --restart=on-failure
    --restart=on-failure:3
    --restart=unless-stopped
```

the `restart=on-failure:3` almost covers the request. The `on-failure:3` restarts 3 times the container if the container exits with a "bad" status code. Instead sometimes one wants to give a limited number of restarts (and then one needs to investigate) to containers that may exit with a good status code (like `exit 0`).

Yes I know, the application should be good enough to exit with a proper status code in case of error, but in the real world not all applications exit with proper status codes. Therefore a `restart=limited:3` could help in this case.

Hopefully it could be implemented without a major effort. Thank you!

**Comment by thaJeztah** [comment]
So, IIUC, the proposal is effectively to have `--restart=always:3` (always restart the container, but maximum 3 times?).



I should add that I'm not sure if we currently want to expand options for restart-policies, as they are already somewhat crossing the line between container _runtime_ and _orchestration_ (and there's some already nasty parts in the current implementation)

**Comment by pier4r** [comment]
Yes. `--restart=always:3` would do it. Because some app crash or exit with a non proper exit code and one wants to restart only so often.

Yes I understand that it goes on the orchestration part. I asked in docker-compose but they, correctly, sent me here. If one uses at most docker-compose as a docker wrapper, which other tool could cover this case without bloating the stack of tools that one needs to use?
